### PR TITLE
Updated Summon Swords and Summon Magic

### PR DESCRIPTION
**Pull request by GeneralVeers25**
What it says on the tin - Summon Swords is reworked to reduce the number of 'useless' categories, such as glass shards and broken bottles. These and a variety of other 'bad' or joke melee weapons (such as the toy banhammer) have been added to the 'Unlucky' category. Summon Swords can also now summon batons, the bullwhip, the morningstar, and cult blades, even though these aren't all explicitly swords.

Summon Magic can no longer summon any artifacts, notably the Staff of Change, Gem-Encrusted Hardsuit, and Scrying Orb. Instead, it now has access to every one-use spellbook in the game, barring the Winter Spellbook, and summons robes for everyone as well so that they may cast their new spells.

:cl:
 * rscadd: Summon Swords can now summon a wider variety of melee weapons and you may get a shield out of the bargain too! All of the 'bad' melee weapons can still be summoned if you are particularly unlucky.
 * rscadd: Summon Guns now summons even more guns.
 * rscadd: Summon Magic can now summon every one-use spellbook in the game and provides everyone with the slick wizard robes they need to cast said spells.
 * rscdel: Summon Magic can no longer summon any artifacts, including the Staff of Change, Gem-Encrusted Hardsuit, or Scrying Orb

### CODE REVIEW

**Comment by SonixApache** [comment]
remove ancient
bundle bunga or bust

**Comment by Shadowmech88** [comment]
I don't think allowing the high frequency katana to be obtained is a good idea.

**Comment by SGuivenne** [comment]
@Shadowmech88 Look, we're already talking all sorts of horrific shit, at this point the holo katana just makes you a target.

**Comment by Caboozles** [comment]
If Summon X is being locked to roundstart, I think this is pretty great, because it essentially cranks the round up to 12 just by existing. Even if weapons are massively overpowered, you may want to be careful because the other guy might have pain mirror or mind swap.

**Comment by Kammerjunk** [comment]
I don't see "all sorts of horrific shit" but it doesn't address that it's a 200 force (??) 100 sharpness (?????) sword that's explicitly an admin-only item. The old comment even says "maybe nerf later."
It would be like putting the actual banhammer weapon in there. Eswords and arm blades don't really match up.

**Comment by Arthurdentist** [comment]
remove ancient

bundle bonus only

**Comment by GeneralVeers25** [comment]
I could knock the probability for the HF Katana down to 1% or lower, but since Armblades and Energy Axes are already in, I don't see why it shouldn't be included.

I don't mind taking out Ancient Magic, but I think it would be fun to give the crew a chance at it because those spells are wild.

**Comment by Arthurdentist** [comment]
I don't really think the crew should get to enjoy those spells without the associated risk. The risk of the spellbook bundle is that you get a bad combination, the reward is the potential or unique one of a kind spells and for 7 spells instead of the usual 5, but adding the ancient spellbook to summon magic makes it a bit more common and a bit less worth it to roll the bundle, cause you might get to try the spells anyway if you are lucky enough during summon magic.

**Comment by Kammerjunk** [comment]
[Armblade](https://github.com/vgstation-coders/vgstation13/blob/d9e848094b6925abf2f76e05e25d483a69edf908/code/game/objects/items/weapons/armblade.dm) says force 30
There's no energy axe. There's the [rusty variant](https://github.com/vgstation-coders/vgstation13/blob/d9e848094b6925abf2f76e05e25d483a69edf908/code/game/objects/items/weapons/melee/energy.dm#L44) of the energy axe, which is just an esword - force 3 when off, force 30 when on
edit: nvm it's in one of the collapsed parts of the changes tab. It shouldn't be included for the same reason e-axe shouldn't which is the thing that it says in the comment - it's retardedly powerful and doesn't compare to anything else

**Comment by GeneralVeers25** [comment]
@Arthurdentist I just checked in the Bundle code, and it looks like all that happens there is functionally the same thing that happens here, a randomized selection from all the one-use spellbooks, with a possibility for duplicates. But you also get 7 chances to roll it, whereas Summon Magic only gives you one (unless you cast it more than once, which is why I still think the price should be changed but that's not relevant here).

I don't mind pulling it out, but for the sake of !!FUN!! why not give it a chance?

@Kammerjunk Well the regular energy axe was already included under a 5% chance in the regular version of summon swords. Most people just don't roll it because the current version is gunked up. I think that they should still be available for that small, chance that RNGesus favors you.

**Comment by GeneralVeers25** [comment]
To put this into perspective, because these are random draws you have a 1 in 18 chance of getting the "Katana" category and then a 5% chance to get the HF Katana. This gives you a 0.277% chance to get the HF Katana - less than half of 1%! For the Energy Axe, it's even slimmer, at 0.14%.

For Ancient Magic meanwhile, it's a 1 in 28 chance to get the Ancient Magic spellbook, or 3.5%, but then each spell has a 25% chance to be selected, for a 0.8% chance to get Disintegrate. And regardless, if you die, the secret magic is lost because anyone else who tries to read your book gets gibbed.

Edit: Because these checks aren't just done once, but can be applied to multiple people at the start of the shift, if there are 20 people on-station when the Summons are cast, there's a 5% chance someone gets the HF Katana, a 2.8% chance for the Energy Axe, and a 16% chance for any of the Ancient Magics. If these numbers are too high, we can certainly drop them down.

**Comment by Caboozles** [comment]
I’d say drop the odds for these items but still make them feasible. I’m more interested in summon spares or Poly-morph being used on station than Ei Nath. We can also do manual trimming if we don’t want Ei Nath.

**Comment by SonixApache** [comment]
just drop the retarded options, man

**Comment by GeneralVeers25** [comment]
I'd have to edit the actual code for the Ancient Spellbook to reduce chances of people rolling Disintegrate and I don't want to do that, so I'll drop these items when I'm home.

**Comment by coldcola** [comment]
was artifacts being summoned as part of Summon Magic unintended? why remove it otherwise?
EDIT TO ADD INFO AS EXPLAINED ON SERB:

"The artifacts are being taken out of summon magic because they flat out don't belong there. The SoC is worth more than pretty much any other spell on the vanilla list, because it's an artifact, because it creates Xenos, etc."
"It's being replaced with a chance to get any other spell."

Who said they don't belong there? Was it originally unintended for them to be produced and left in?
I have never fought for the SOC to remain in over any other specific artifact, but again - why are we removing all of them because one is a little much?
Are you really 'replacing' it with a chance to get any other spell or are you removing functionality and leaving something functioning as it always did otherwise? What spells have you added to the oneuse spellbooks list?

**Comment by GeneralVeers25** [comment]
Presumably, Summon Magic was written back when only one Wizard would spawn. Keeping the artifacts was probably intentional back when it was added, but I don't think that taking the artifacts out reduces the functionality of the spell at all. If anything the artifacts detract from the spell, since the scrying orb and hardsuit are just memes. Only the Staff of Change was worth anything because it was so much more destructive than any other spell on the list and can pretty much invalidate every other effect of Summon Magic because anyone hit with the Staff of Change loses their spells!

We have one-use spell books for every spell that has been added to the Wizard's spellbook in the code. I expanded the possible list that Summon Magic can draw from to include all of these spellbooks, with the exception of the Ancient Magic spellbooks, to replace the artifacts. That's 15 new spells being added to contrast with four items going away (summon robes, which was superfluous since the crew gets them, and the three artifacts.)

If we're going to keep artifacts in Summon Magic, we should add the rest of them too, but even then the SoC should go away regardless, and I don't see why that should happen now. Maybe in the future.

**Comment by coldcola** [comment]
SOC being removed is fair, but I disagree that removing the orb and hardsuit because they are "memes" is the right move.  How are they any more memes than some of the spellbooks - especially considering you can not roll survivor and end up with a harmful 'meme' spell that you can't use without a bwoink?

How have you added "15 new spells" when those spells were already being pulled properly by Summon Magic? Best I can see is that you've added the ancient spellbook.

**Comment by GeneralVeers25** [comment]
> How have you added "15 new spells" when those spells were already being pulled properly by Summon Magic? Best I can see is that you've added the ancient spellbook.

Are you trolling me?

Edit: let me be more specific, before Summon Magic only called from the following spells: "fireball","smoke","blind","mindswap","forcewall","knock","horsemask","blink","disorient","staffchange","armor","scrying", "clowncurse", "mimecurse", "shoesnatch", "robesummon"

I added 15 to this list, taking out scrying, armor, robesummon, and staffchange.

**Comment by coldcola** [comment]
I didn't scroll to the right, you're right I was confusing the spellbook bundle and Summon Magic again.

Still don't like removal of all artifacts :/

**Comment by Lunstra** [comment]
Perhaps the hardsuit could possibly be left in as a low chance roll instead of robes?

**Comment by GeneralVeers25** [comment]
Perhaps. I'd like to come back to this in the future and consider adding some of the 'weaker' staffs, such as Animation and Force if Summon Magic just seems weak. But right now, I think that the expanded spell list deserves a run and we can talk about more changes after we see how this plays out.

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 1]
```diff
@@ -127,43 +129,68 @@
...
 		if("templar")
 			new /obj/item/clothing/suit/armor/knight/templar(get_turf(src))
 			new /obj/item/clothing/head/helmet/knight/templar(get_turf(src))
+			if(prob(50))
+				new /obj/item/weapon/shield/riot/buckler(get_turf(src))
+		if("roman")
+			new /obj/item/clothing/under/roman(get_turf(src))
+			new /obj/item/clothing/head/helmet/roman(get_turf(src))
+			new /obj/item/clothing/shoes/roman(get_turf(src))
+			new /obj/item/weapon/shield/riot/roman(get_turf(src)) //guaranteed shield to make up for the lack of armor. Also fits the theme better.
```
Comment (SonixApache) [inline code/game/gamemodes/wizard/rightandwrong.dm:179]: Don't tell anyone, but roman clothes are actually pretty robust

Comment (GeneralVeers25) [reply code/game/gamemodes/wizard/rightandwrong.dm:179]: Roma Invicta

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 2]
```diff
@@ -43,7 +43,7 @@
 			return equip_guns()
 
 /mob/living/carbon/human/proc/equip_guns()
-	var/randomizeguns = pick("taser","stunrevolver","egun","laser","retro","laserak","revolver","detective","c20r","nuclear","deagle","gyrojet","pulse","silenced","cannon","doublebarrel","shotgun","combatshotgun","mateba","smg","uzi","microuzi","crossbow","saw","hecate","osipr","gatling","bison","ricochet","spur","nagant","obrez","beegun","beretta","usp","glock","luger","colt","plasmapistol")
+	var/randomizeguns = pick("taser","stunrevolver","egun","laser","retro","laserak","revolver","detective","c20r","nuclear","deagle","gyrojet","pulse","silenced","cannon","doublebarrel","shotgun","combatshotgun","mateba","smg","uzi","microuzi","crossbow","saw","hecate","osipr","gatling","bison","ricochet","spur","nagant","obrez","beegun","beretta","usp","glock","luger","colt","plasmapistol","plasmarifle", "ion", "bulletstorm", "cannon", "laserpistol", "siren", "lawgiver")
```
Comment (GeneralVeers25) [inline code/game/gamemodes/wizard/rightandwrong.dm]: Just noticed that "cannon" is duplicated here for both the Combustion Cannon and Laser Cannon. I will fix this when I'm home.

**Review by GeneralVeers25 — COMMENTED** [review]
Accidentally had two different "cannon" options. I've corrected that.

Edit: As an addendum, just realized I can code from my day job. Pretty sweet.

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 3]
```diff
@@ -127,43 +129,68 @@
...
 			new /obj/item/clothing/suit/armor/knight/red(get_turf(src))
 			new /obj/item/clothing/head/helmet/knight/red(get_turf(src))
+			if(prob(50))
+				new /obj/item/weapon/shield/riot/buckler(get_turf(src))
 		if("templar")
 			new /obj/item/clothing/suit/armor/knight/templar(get_turf(src))
 			new /obj/item/clothing/head/helmet/knight/templar(get_turf(src))
+			if(prob(50))
+				new /obj/item/weapon/shield/riot/buckler(get_turf(src))
+		if("roman")
```
Comment (jknpj) [inline code/game/gamemodes/wizard/rightandwrong.dm:175]: MONTHLY REMINDER THAT OUTFIT DATUMS COULD PREVENT THIS HELL

**Comment by MadmanMartian** [comment]
This okay to go?

**Pull request merged**